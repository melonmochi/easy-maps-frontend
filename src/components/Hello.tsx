import * as React from 'react';
import { Button } from 'antd';

interface HelloProps {
  compiler: string;
  framework: string;
}

const Hello = (props: HelloProps) => (
  <div>
    <Button type="primary">Primary</Button>
    <h1>
      Hello from {props.compiler} and {props.framework}!
    </h1>
  </div>
);

export default Hello;
