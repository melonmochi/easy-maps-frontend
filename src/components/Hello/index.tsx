import * as React from 'react';
import { Button } from 'antd';
import './style';

interface HelloProps {
  compiler: string;
  framework: string;
}

const Hello = (props: HelloProps) => (
  <div>
    <Button className="hello-button" type="primary">
      Primary
    </Button>
    <h1>
      Hello from {props.compiler} and {props.framework}!
    </h1>
  </div>
);

export default Hello;
