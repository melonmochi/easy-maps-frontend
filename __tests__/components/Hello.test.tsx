import * as React from 'react';
import { shallow } from 'enzyme';
import { Hello } from 'components';

const props = {
  compiler: 'Typescript',
  framework: 'React',
}

test('Hello matches the snapshot', () => {
  const hello = shallow(<Hello {...props} />);
  // Snapshot demo
  expect(hello).toMatchSnapshot();
});
