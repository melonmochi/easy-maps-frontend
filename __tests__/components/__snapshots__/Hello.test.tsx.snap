// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello matches the snapshot 1`] = `
<div>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    type="primary"
  >
    Primary
  </Button>
  <h1>
    Hello from 
    Typescript
     and 
    React
    !
  </h1>
</div>
`;
